swaylock/main.c

```diff
@@ -101,12 +102,17 @@ void notify_key(enum wl_keyboard_key_state state, xkb_keysym_t sym, uint32_t cod
 			if (verify_password()) {
 				exit(0);
 			}
-			password = malloc(1024); // TODO: Let this grow
+			password_size = 1024;
+			password = malloc(password_size);
 			password[0] = '\0';
 			break;
 		default:
 			{
 				int i = strlen(password);
+				if (i + 1 == password_size) {
+					password_size += 1024;
+					password = realloc(password, password_size);
```

Comment:
Better than the old way of crashing when you entered too much text and buffer overflowed, which I imagine could lead to code execution if done by a crafty person.